src/freenet/node/PeerManager.java

```diff
@@ -238,58 +238,60 @@ private boolean readPeers(File peersFile, NodeCrypto crypto, OpennetManager open
-					continue;
-				} catch(PeerParseException e2) {
-					Logger.error(this, "Could not parse peer: " + e2 + '\n' + fs.toString(), e2);
-					System.err.println("Cannot parse a friend from the peers file: "+e2);
-					someBroken = true;
-					continue;
-				} catch(ReferenceSignatureVerificationException e2) {
-					Logger.error(this, "Could not parse peer: " + e2 + '\n' + fs.toString(), e2);
-					System.err.println("Cannot parse a friend from the peers file: "+e2);
-					someBroken = true;
-					continue;
-				} catch (RuntimeException e2) {
-					Logger.error(this, "Could not parse peer: " + e2 + '\n' + fs.toString(), e2);
-					System.err.println("Cannot parse a friend from the peers file: "+e2);
-					someBroken = true;
-					continue;
-					// FIXME tell the user???
-				} catch (PeerTooOldException e) {
-				    if(crypto.isOpennet()) {
-				        // Ignore.
-				        Logger.error(this, "Dropping too-old opennet peer");
-				    } else {
-				        // A lot more noisy!
-				        droppedOldPeers.add(e, fs.get("myName"));
-				    }
-                    someBroken = true;
-                    continue;
-                }
+		List<SimpleFieldSet> peerEntries = new ArrayList<>();
+		// read the peers file
+		try {
+			while (true) {
+			  peerEntries.add(new SimpleFieldSet(br, false, true));
 			}
 		} catch(EOFException e) {
 			// End of file, fine
 		} catch(IOException e1) {
 			Logger.error(this, "Could not read peers file: " + e1, e1);
 		}
+
+		List<PeerNode> createdNodes = new ArrayList<>();
+		for (SimpleFieldSet fs : peerEntries) {
+			try {
+				createdNodes.add(PeerNode.create(fs, node, crypto, opennet, this));
+			} catch(FSParseException e2) {
+				Logger.error(this, "Could not parse peer due to broken fieldset syntax: " + e2 + '\n' + fs.toString(), e2);
+				System.err.println("Cannot parse a friend from the peers file due to broken fieldset syntax: "+e2);
+				someBroken = true;
+			} catch(PeerParseException e2) {
+				Logger.error(this, "Could not parse peer: " + e2 + '\n' + fs.toString(), e2);
+				System.err.println("Cannot parse a friend from the peers file: "+e2);
+				someBroken = true;
+			} catch(ReferenceSignatureVerificationException e2) {
+				Logger.error(this, "Could not verify signature of peer: " + e2 + '\n' + fs.toString(), e2);
+				System.err.println("Cannot verify signature of a friend from the peers file: "+e2);
+				someBroken = true;
+			} catch (RuntimeException e2) {
+				Logger.error(this, "Could not parse peer: " + e2 + '\n' + fs.toString(), e2);
+				System.err.println("Cannot parse a friend from the peers file: " + e2);
+				someBroken = true;
+				// FIXME tell the user???
+			}  catch (PeerTooOldException e) {
+				someBroken = true;
+				if(crypto.isOpennet()) {
+					// Ignore.
+					Logger.error(this, "Dropping too-old opennet peer");
+				} else {
+					// A lot more noisy
+					droppedOldPeers.add(e, fs.get("myName"));
+				}
+			}
+		}
+
+		for (PeerNode pn : createdNodes) {
```

Comment:
Even if this whole block is only copied, please add braces for all parts of `if` statements. 🙂

Reply:
Braces and spaces added (and some tabify) ☺